manifests/plugin/python.pp

```diff
@@ -20,26 +15,39 @@
   }
   # End deprecation
 
-  file {
-    "${name}.load":
-      ensure  => $ensure,
-      path    => "${conf_dir}/${order}-${name}.conf",
-      owner   => 'root',
-      group   => $collectd::params::root_group,
-      mode    => '0640',
-      content => template('collectd/python.conf.erb'),
-      notify  => Service['collectd'],
+  $ensure_modulepath = $ensure ? {
+      'absent' => $ensure,
+      default  => 'directory',
   }
 
-  file {
-    "${name}.script":
-      ensure  => $ensure,
-      path    => "${modulepath}/${module}.py",
-      owner   => 'root',
+  file { $modulepath :
+      ensure  => $ensure_modulepath,
+      mode    => '0750',
+      owner   => root,
       group   => $collectd::params::root_group,
-      mode    => '0640',
-      source  => $script_source,
-      require => File["${name}.load"],
-      notify  => Service['collectd'],
+      require => Package[$collectd::params::package]
+  }
+
+  collectd::plugin { 'python' :
+      ensure   => $ensure,
+      interval => $interval,
```

Comment:
By default its inherited from collectd, but it is pretty fine to specify a completely different interval. And depending on what you do you in your python module you might want completely different intervals from the default.